src/pages/iou/request/step/IOURequestStepDistance.tsx

```diff
@@ -78,7 +79,18 @@ function IOURequestStepDistance({
             },
         [optimisticWaypoints, transaction],
     );
```

Comment:
@wildan-m ⚠️ I noticed an issue which happens only when the RHP is dismissed for the first time -> the map component in the report resets / reloads. This doesn't happen if we open Distance RHP again and dismiss for the 2nd time.

https://github.com/user-attachments/assets/cc80f95d-fad1-4be3-9645-5a45423a1af6

I'm pretty sure that this will be reported as regression if we ship it this way, because since we fixed the issue in the sense that the map does show up now when dismissing the RHP, we now have that flicker map reset / reload on first time RHP dismissal.

Please look into this and see if you can figure out a way to not have the report map reset / reload when the RHP is dismissed for the first time and instead keep it the way it looks when it loads for the first time while RHP is still open if possible.

Reply:
@ikevin127 After further investigation, this issue related to my previous mentioned issue https://github.com/Expensify/App/issues/48630#issuecomment-2438017630 where `transactionBackup` is not exactly the same with unmodified `transaction` data.

For instance, in backup, the pendingFields is not removed, this will cause incorrect result here:

https://github.com/Expensify/App/blob/652d2ffa1b099f9c5a1f0cdce780792097f736ad/src/components/ReportActionItem/MoneyRequestView.tsx#L258-L260

I think the only way to resolve that flicker issue and other fields flicker is to make the backup data exactly the same with unmodified transaction data when they are online, and it will require BE change.

@Gonals is it possible that when we call `GetRouteForBackup` we also get the same unmodified transaction data when they are online?

The missing fields can be seen in previous comment https://github.com/Expensify/App/issues/48630#issuecomment-2438017630

Reply:
@wildan-m Thanks for the investigation and explanation, I just wanted to point it out and ask @Gonals whether we should fix it now or in another issue.

Reply:
I found another issue where the generated map is not clickable due to pendingFields and pendingAction not being removed. Making 'transactionBackup' identical to unmodified 'transaction' when online is crucial to prevent flickering and an unclickable map, but I'm unsure if this is possible from the backend perspective.

Reply:
I think this has a different root cause, right? Since it requires BE changes, I think it is a different bug

Reply:
> I think this has a different root cause, right? Since it requires BE changes, I think it is a different bug

@Gonals I agree that flickering is not directly related to the original issue and fixing it is not straightforward.